server/core/src/initServer.cpp

```diff
@@ -723,121 +723,12 @@ int initRsCommWithStartupPack(rsComm_t* rsComm, startupPack_t* startupPack, bool
-        }
-        rstrcpy( rsComm->proxyUser.userName, tmpStr, NAME_LEN );
-        if ( strcmp( tmpStr, PUBLIC_USER_NAME ) == 0 ) {
-            rsComm->proxyUser.authInfo.authFlag = PUBLIC_USER_AUTH;
-        }
-
-        tmpStr = getenv( SP_PROXY_RODS_ZONE );
-        if ( tmpStr == NULL ) {
-            rodsLog( LOG_NOTICE,
-                    "initRsCommWithStartupPack: env %s does not exist",
-                    SP_PROXY_RODS_ZONE );
-            return SYS_GETSTARTUP_PACK_ERR;
-        }
-        rstrcpy( rsComm->proxyUser.rodsZone, tmpStr, NAME_LEN );
-
-        tmpStr = getenv( SP_CLIENT_USER );
-        if ( tmpStr == NULL ) {
-            rodsLog( LOG_NOTICE,
-                    "initRsCommWithStartupPack: env %s does not exist",
-                    SP_CLIENT_USER );
-            return SYS_GETSTARTUP_PACK_ERR;
-        }
-        rstrcpy( rsComm->clientUser.userName, tmpStr, NAME_LEN );
-        if ( strcmp( tmpStr, PUBLIC_USER_NAME ) == 0 ) {
-            rsComm->clientUser.authInfo.authFlag = PUBLIC_USER_AUTH;
-        }
-
-        tmpStr = getenv( SP_CLIENT_RODS_ZONE );
-        if ( tmpStr == NULL ) {
-            rodsLog( LOG_NOTICE,
-                    "initRsCommWithStartupPack: env %s does not exist",
-                    SP_CLIENT_RODS_ZONE );
-            return SYS_GETSTARTUP_PACK_ERR;
-        }
-        rstrcpy( rsComm->clientUser.rodsZone, tmpStr, NAME_LEN );
-
-        tmpStr = getenv( SP_REL_VERSION );
-        if ( tmpStr == NULL ) {
-            rodsLog( LOG_NOTICE,
-                    "getstartupPackFromEnv: env %s does not exist",
-                    SP_REL_VERSION );
-            return SYS_GETSTARTUP_PACK_ERR;
-        }
-        rstrcpy( rsComm->cliVersion.relVersion, tmpStr, NAME_LEN );
-
-        tmpStr = getenv( SP_API_VERSION );
-        if ( tmpStr == NULL ) {
-            rodsLog( LOG_NOTICE,
-                    "initRsCommWithStartupPack: env %s does not exist",
-                    SP_API_VERSION );
-            return SYS_GETSTARTUP_PACK_ERR;
-        }
-        rstrcpy( rsComm->cliVersion.apiVersion, tmpStr, NAME_LEN );
-
-        tmpStr = getenv( SP_OPTION );
-        if ( tmpStr == NULL ) {
-            rodsLog( LOG_NOTICE,
-                    "initRsCommWithStartupPack: env %s does not exist",
-                    SP_OPTION );
-        }
-        else {
-            rstrcpy( rsComm->option, tmpStr, LONG_NAME_LEN );
-        }
 
-        if (std::getenv(irods::RODS_CS_NEG)) {
-            require_cs_neg = true;
+        // Set the SP_OPTION environment variable so that server-to-server connections
+        // forward the name of the connected client to remote servers. This information
+        // is used by the rcConnect library when establishing a connection to a server.
+        if (setenv(SP_OPTION, rsComm->option, /* overwrite */ 1) != 0) {
+            log_agent::warn("{}: Could not set environment variable [{}] for ips.", __func__, SP_OPTION);
```

Comment:
should we log the attempted `rsComm->option` value on failure as well?

Reply:
I think that's what this line is doing. I feel I'm misunderstanding the question.

Reply:
Oh oh, I see now. You're talking about logging the value of `->option`.

Reply:
Sure, that sounds like a good idea.

Reply:
right, in case that's another clue for the admin.

Reply:
Done.